test/e2e/service_bootstrap.py

```diff
@@ -22,102 +22,44 @@
 import subprocess
 
 from acktest import resources
+from acktest.bootstrapping import Resources, BootstrapFailureException
+from acktest.bootstrapping.iam import Role
+from acktest.bootstrapping.s3 import Bucket
 from acktest.aws.identity import get_region, get_account_id
 from e2e import bootstrap_directory
-from e2e.bootstrap_resources import TestBootstrapResources
 from acktest.aws.s3 import duplicate_bucket_contents
 from e2e.bootstrap_resources import TestBootstrapResources, SAGEMAKER_SOURCE_DATA_BUCKET
 
 
-def service_bootstrap() -> dict:
+def service_bootstrap() -> Resources:
     logging.getLogger().setLevel(logging.INFO)
 
-    scalable_table = create_dynamodb_table("ScalableDynamoTableName")
-    registered_table = create_dynamodb_table("RegisteredDynamoTableName")
-
-    if not wait_for_dynamodb_table_active(
-        scalable_table
-    ) or not wait_for_dynamodb_table_active(registered_table):
-        raise Exception("DynamoDB tables did not become ACTIVE")
-
-    return TestBootstrapResources(
-        create_data_bucket(),
-        create_execution_role(),
-        scalable_table,
-        register_scalable_dynamodb_table(registered_table),
-    ).__dict__
-
-def write_bootstrap_config_entry(key: str, value: str, bootstrap_file_name: str = "bootstrap.yaml"):
-    bootstrap_file = bootstrap_directory / bootstrap_file_name
-    config = {key: value}
-
-    if os.path.isfile(bootstrap_file):
-        stream = resources.read_bootstrap_config(bootstrap_directory)
-        config.update(stream)
-    
-    resources.write_bootstrap_config(config, bootstrap_directory)
-
-def create_execution_role() -> str:
     region = get_region()
-    role_name = resources.random_suffix_name(f"ack-sagemaker-execution-role", 63)
-    iam = boto3.client("iam", region_name=region)
-
-    iam.create_role(
-        RoleName=role_name,
-        AssumeRolePolicyDocument=json.dumps(
-            {
-                "Version": "2012-10-17",
-                "Statement": [
-                    {
-                        "Effect": "Allow",
-                        "Principal": {"Service": "sagemaker.amazonaws.com"},
-                        "Action": "sts:AssumeRole",
-                    }
-                ],
-            }
+    account_id = get_account_id()
+    bucket_name = f"ack-test-bucket-{region}-{account_id}"
+
+    resources = TestBootstrapResources(
+        DataBucket=Bucket(bucket_name),
+        SageMakerExecutionRole=Role(
+            "ack-sagemaker-execution-role",
+            "sagemaker.amazonaws.com",
+            managed_policies=[
+                "arn:aws:iam::aws:policy/AmazonSageMakerFullAccess",
```

Comment:
Is sagemaker and s3 fullaccess sufficient for autoscaling? Looks like that was what was there previously, but double-checking.

Reply:
yes. SageMakerFullAccess has all the permissions reqd for autoscaling